mm2src/mm2_main/tests/mm2_tests/mm2_tests_inner.rs

```diff
@@ -7927,3 +7950,342 @@ fn test_eth_swap_negotiation_fails_maker_no_fallback() {
+    let req: InitHwRequest = serde_json::from_value(json!({ "device_pubkey": null })).unwrap();
+    let res = match block_on(init_trezor(ctx.clone(), req)) {
+        Ok(res) => res,
+        _ => {
+            panic!("cannot init trezor");
+        },
+    };
+
+    block_on(async {
+        loop {
+            let status_req = RpcTaskStatusRequest {
+                task_id: res.task_id,
+                forget_if_finished: false,
+            };
+            match init_trezor_status(ctx.clone(), status_req).await {
+                Ok(res) => {
+                    log!("trezor init status={:?}", serde_json::to_string(&res).unwrap());
+                    match res {
+                        RpcTaskStatus::Ok(_) => {
+                            log!("device initialized");
+                            break;
+                        },
+                        RpcTaskStatus::Error(_) => {
+                            log!("device in error state");
+                            break;
+                        },
+                        RpcTaskStatus::InProgress(_) => log!("trezor init in progress"),
+                        RpcTaskStatus::UserActionRequired(_) => log!("device is waiting for user action"),
+                    }
+                },
+                _ => {
+                    panic!("cannot get trezor status");
+                },
+            };
+            Timer::sleep(5.).await
+        }
+    });
+    ctx
+}
+
+/// Tool to run withdraw directly with trezor device or emulator (no rpc version, added for easier debugging)
+/// run cargo test with '--ignored' option
+/// to use trezor emulator add '--features trezor-udp' option to cargo test params
+#[test]
+#[ignore]
+#[cfg(all(not(target_arch = "wasm32")))]
+fn test_withdraw_from_trezor_segwit_no_rpc() {
+    use common::wait_until_ms;
+
+    let ticker = "tBTC-segwit";
+    let coin_conf = json!({
+        "coin": ticker,
+        "name": "tbitcoin",
+        "fname": "tBitcoin",
+        "rpcport": 18332,
+        "pubtype": 111,
+        "p2shtype": 196,
+        "wiftype": 239,
+        "segwit": true,
+        "bech32_hrp": "tb",
+        "address_format": {
+            "format": "segwit"
+        },
+        "orderbook_ticker": "tBTC",
+        "txfee": 0,
+        "estimate_fee_mode": "ECONOMICAL",
+        "mm2": 1,
+        "is_testnet": true,
+        "required_confirmations": 0,
+        "protocol": {
+            "type": "UTXO"
+        },
+        // added to params from coins
+        "derivation_path": "m/84'/1'",
+        "trezor_coin": "Testnet",
+    });
+    let mm_conf = json!({ "coins": [coin_conf] });
+
+    let ctx = mm_ctx_with_trezor(mm_conf);
+    let priv_key_policy = PrivKeyBuildPolicy::Trezor;
+    let servers: Vec<_> = [
+        "electrum1.cipig.net:10068",
+        "electrum2.cipig.net:10068",
+        "electrum3.cipig.net:10068",
+    ]
+    .iter()
+    .map(|server| json!({ "url": server }))
+    .collect();
```

Comment:
You can also use tbtc electrums constant here, it makes it easier to change electrums for all tests if some of them are down.

Reply:
done